packages/garbo/src/resources/worksheds.ts

```diff
@@ -157,3 +163,62 @@ export function grabMedicine(): void {
   }
   if (handlingChoice()) visitUrl("main.php");
 }
+
+// Function to calculate the value for each item
+function calculateItemValue(item: Item, effect: Effect): number {
+  return (
+    new Potion($item`diabolic pizza`, {
+      effect: effect,
+      duration: Math.sqrt(autosellPrice(item)),
+    }).gross(copyTargetCount()) -
+    mallPrice(item) +
+    (toSlot(item) === $slot`familiar`
+      ? mallPrice($item`box of Familiar Jacks`)
+      : 0) +
+    item.name.length / 10
+  );
+}
+
+// Consolidated function to get the best item for a given letter
+function bestPizzaItemForLetter(letter: string, effect: Effect): Item {
+  const items = $items.all().filter((i) => i.name.startsWith(letter));
+
+  const itemsWithValues = items.map((item) => ({
+    item,
+    value: calculateItemValue(item, effect),
+  }));
+
+  return maxBy(itemsWithValues, (entry) => entry.value).item;
```

Comment:
```suggestion
  return maxBy(itemsWithValues, "value").item;
```

Reply:
Also fwiw if you aren't using the values outside of the maxby it makes more sense to not do the map at all and to instead just ```typescript
return maxBy(items, (it) => calculateItemValue(it, effect));
```